elastic4s-http/src/main/scala/com/sksamuel/elastic4s/http/index/alias/IndexAliasImplicits.scala

```diff
@@ -1,47 +1,46 @@
 package com.sksamuel.elastic4s.http.index.alias
 
-import com.sksamuel.elastic4s.JsonFormat
 import com.sksamuel.elastic4s.alias.{GetAliasDefinition, IndicesAliasesRequestDefinition}
-import com.sksamuel.elastic4s.http.HttpExecutable
 import com.sksamuel.elastic4s.http.index.IndicesAliasResponse
+import com.sksamuel.elastic4s.http.{HttpExecutable, ResponseHandler}
 import org.apache.http.entity.{ContentType, StringEntity}
-import org.elasticsearch.client.{ResponseException, RestClient}
+import org.elasticsearch.client.{Response, ResponseException, RestClient}
 
 import scala.concurrent.Future
 import scala.util.{Failure, Try}
-import scala.collection.JavaConverters._
 
 trait IndexAliasImplicits {
 
   type GetAliasResponse = Map[String, Map[String, Map[String, AnyRef]]]
 
   implicit object GetAliasExecutable extends HttpExecutable[GetAliasDefinition, GetAliasResponse] {
     override def execute(client: RestClient,
-                         request: GetAliasDefinition,
-                         format: JsonFormat[GetAliasResponse]): Future[GetAliasResponse] = {
+                         request: GetAliasDefinition): Future[GetAliasResponse] = {
       val indexPathElement = if (request.indices.isEmpty) "" else s"/${request.indices.mkString(",")}"
       val endpoint = s"$indexPathElement/_alias/${request.aliases.mkString(",")}"
       logger.debug(s"Connecting to $endpoint for get alias request")
 
-      executeAsyncAndMapResponse(client.performRequestAsync("GET", endpoint, _), format, {
-        case re: ResponseException if re.getResponse.getStatusLine.getStatusCode == 404 => Try(Map.empty)
-        case e => Failure(e)
+      client.future("GET", endpoint, Map.empty, new ResponseHandler[GetAliasResponse] {
```

Comment:
Because of #824. 

I didn't have a strong preference either way. I just went with `Map.empty` because you said "I prefer a valid response with an empty Map." The other option is of course going with a failed Future.